src/elements/content-sidebar/activity-feed/annotations/AnnotationActivity.js

```diff
@@ -0,0 +1,246 @@
+// @flow
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import noop from 'lodash/noop';
+import { FormattedMessage } from 'react-intl';
+import TetherComponent from 'react-tether';
+import Avatar from '../Avatar';
+import Media from '../../../../components/media';
+import { MenuItem } from '../../../../components/menu';
+import IconTrash from '../../../../icons/general/IconTrash';
+import IconPencil from '../../../../icons/general/IconPencil';
+import { ACTIVITY_TARGETS } from '../../../common/interactionTargets';
+import DeleteConfirmation from '../common/delete-confirmation';
+import ActivityTimestamp from '../common/activity-timestamp';
+import UserLink from '../common/user-link';
+import ActivityError from '../common/activity-error';
+import ActivityMessage from '../common/activity-message';
+import AnnotationActivityItem from './AnnotationActivityItem';
+import CommentForm from '../comment-form';
+import { bdlGray } from '../../../../styles/variables';
+import { PLACEHOLDER_USER } from '../../../../constants';
+import messages from './messages';
+import type { AnnotationRegionTarget } from './types';
+import type { GetAvatarUrlCallback, GetProfileUrlCallback } from '../../../common/flowTypes';
+import type { Translations } from '../../flowTypes';
+import type { SelectorItems, User } from '../../../../common/types/core';
+import type { BoxCommentPermission, ActionItemError } from '../../../../common/types/feed';
+
+import './AnnotationActivity.scss';
+
+type Props = {
+    created_at: string | number,
+    created_by: User,
+    currentUser?: User,
+    error?: ActionItemError,
+    getAvatarUrl: GetAvatarUrlCallback,
+    getMentionWithQuery?: Function,
+    getUserProfileUrl?: GetProfileUrlCallback,
+    id: string,
+    isActive?: boolean,
+    isDisabled?: boolean,
+    isPending?: boolean,
+    mentionSelectorContacts?: SelectorItems<>,
+    modified_at?: string | number,
+    onDelete?: ({ id: string, permissions?: BoxCommentPermission }) => any,
+    onEdit?: (id: string, text: string, hasMention: boolean, permissions?: BoxCommentPermission) => any,
+    onSelect?: (id: string) => any,
+    permissions?: BoxCommentPermission,
+    tagged_message?: string,
```

Comment:
Is this up-to-date with the latest schema?

Reply:
Watch out, the Comment API does some data transformation you may need to be aware of (esp if annotations is different):

https://github.com/box/box-ui-elements/blob/master/src/api/Comments.js#L53-L58